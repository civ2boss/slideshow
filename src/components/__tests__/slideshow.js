import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Slideshow from '../Slideshow';
import App from '../App';

test('Slideshow renders without crashing', () => {
  global.fetch = jest.fn().mockImplementation(() => {
    return new Promise((res, rej) => {
      res({
        ok: true,
        Id: '123',
        json: function() {
          return {
            photos: {
              photo: [
                {
                  farm: 5,
                  id: '39027851635',
                  isfamily: 0,
                  isfriend: 0,
                  ispublic: 1,
                  owner: '70152224@N08',
                  secret: 'f8bd038f57',
                  server: '4757',
                  title: 'Liam. Day Fourty-Four.'
                },
                {
                  farm: 5,
                  id: '39215590324',
                  isfamily: 0,
                  isfriend: 0,
                  ispublic: 1,
                  owner: '70152224@N08',
                  secret: 'fa0c4bd546',
                  server: '4724',
                  title: 'Liam. Day Fourty-Three.'
                }
              ]
            }
          };
        }
      });
    });
  });
  const app = shallow(<App />).instance();

  const component = renderer.create(
    <Slideshow
      selected={app.state.selected}
      photos={app.state.photos}
      flickrPhotoUrl={app.flickrPhotoUrl}
      selectPhoto={app.selectPhoto}
      prevPhoto={app.prevPhoto}
      nextPhoto={app.nextPhoto}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
