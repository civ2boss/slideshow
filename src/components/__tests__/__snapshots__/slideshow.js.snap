// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slideshow renders without crashing 1`] = `
<div
  className="slideshow"
>
  <div
    className="viewer"
  >
    <div
      className="image"
    >
      <img
        alt=""
        src="https://farmundefined.staticflickr.com/undefined/undefined_undefined_c.jpg"
      />
    </div>
    <div
      className="arrow arrow-left"
    >
      <a
        href=""
        onClick={[Function]}
      >
        &lt;
      </a>
    </div>
    <div
      className="arrow arrow-right"
    >
      <a
        href=""
        onClick={[Function]}
      >
        &gt;
      </a>
    </div>
  </div>
  <div
    className="thumbnails"
  >
    <ul />
  </div>
</div>
`;
